core/embed/rust/src/micropython/gc.rs

```diff
@@ -82,3 +109,9 @@ impl<T: ?Sized> Deref for Gc<T> {
         unsafe { self.0.as_ref() }
     }
 }
+
+impl<T: ?Sized> DerefMut for Gc<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { self.0.as_mut() }
+    }
+}
```

Comment:
@matejcik @jpochyla do we rely on `Gc<T>` being immutable anywhere in our code?

Reply:
Implicitly in most places, I believe. In particular `Gc<>` does not give us any sort of guarantee whatsoever on whether the same thing is mutated elsewhere. That's why the recommended way to do this is `unsafe { Gc::<T>::as_mut(val) }`. This `DerefMut` does not do anything to guarantee safety of that operation.

Reply:
Let's drop it and use `unsafe { Gc::<T>::as_mut(val) }` then.

Or we can add something like `Gc<[T]>::new_slice_init(len: usize, initFn: impl FnOnce(&mut [T])) -> Result<Self, Error>`, not sure if it would be useful outside of this case.

Reply:
changed: https://github.com/trezor/trezor-firmware/commit/c3f6bf1a55016d1c934865ffc0bf21d842b62723